libpod/container_inspect.go

```diff
@@ -196,7 +197,10 @@ func (c *Container) getContainerInspectData(size bool, driverData *define.Driver
 			// An error here is not considered fatal; no health state will be displayed
 			logrus.Error(err)
 		} else {
-			data.State.Health = healthCheckState
+			if reflect.DeepEqual(healthCheckState, define.HealthCheckResults{}) {
+				data.State.Health = &healthCheckState
+			}
+			data.State.Health = nil
```

Comment:
This looks wrong,  you set it and then unset it. Should this be in a else statement?